packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -636,4 +646,49 @@ export class ActionExecutionContext
 	async hackGetMediaObjectDuration(mediaId: string): Promise<number | undefined> {
 		return getMediaObjectDuration(this._context, mediaId, this.studioId)
 	}
+
+	async listPeripheralDevices(): Promise<(PeripheralDevicePublic & { actions: SubdeviceAction[] | undefined })[]> {
+		const parentDeviceIds = this._cache.PeripheralDevices.findAll(
+			(doc: PeripheralDevice) =>
+				doc.studioId === this._context.studioId && doc.type === PeripheralDeviceType.PLAYOUT
+		).map((doc) => doc._id)
+		if (parentDeviceIds.length === 0) {
+			throw new Error('No parent devices are configured')
+		}
+
+		const devices = await this._context.directCollections.PeripheralDevices.findFetch({
+			parentDeviceId: {
+				$in: parentDeviceIds,
+			},
+		})
+
+		return devices.map((d) => {
+			// Only expose a subset of the PeripheralDevice to the blueprints
+			return literal<Complete<PeripheralDevicePublic & { actions: SubdeviceAction[] | undefined }>>({
```

Comment:
should this `PeripheralDevicePublic & { actions: SubdeviceAction[] | undefined }` type be named and user in the interface definition?